src/Root.tsx

```diff
@@ -48,6 +49,27 @@ export async function onLatestVersionChange(newVer: SemVer | null | undefined, o
 const Root = observer(() => {
   const { app, modals } = getRootStore();
 
+  React.useEffect(() => {
+    const searchParams = new URLSearchParams(document.location.search);
+    const result = searchParams.get("Diagram");
+    if (result === null) return;
+
+    // Replace URL-safe characters with base64 equivalents
+    let base64String = result.replace(/-/g, "+").replace(/_/g, "/");
```

Comment:
Why this is needed?
Why replacing characters is needed before calling `window.atob`, but is not needed before calling `window.btoa`?
Did you miss one of them? Please explain the code and double-check

Reply:
Since '+' and '/' is not url friendly, i have change them to '-' and '_', it seems like I forget to add the process in the btoa side, I have added it back to it. Thank you for the reminder.